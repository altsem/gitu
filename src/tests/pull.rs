use super::*;

#[test]
fn pull_menu_no_remote_or_upstream_set() {
    let ctx = TestContext::setup_clone();
    run(ctx.dir.path(), &["git", "branch", "--unset-upstream"]);
    snapshot!(ctx, "F");
}

#[test]
fn pull_menu_existing_push_remote_and_upstream() {
    let ctx = TestContext::setup_clone();
    run(
        ctx.dir.path(),
        &["git", "config", "branch.main.pushRemote", "origin"],
    );
    snapshot!(ctx, "F");
}

#[test]
fn pull_upstream() {
    let ctx = TestContext::setup_clone();
    clone_and_commit(&ctx.remote_dir, "remote-file", "hello");
    snapshot!(ctx, "Fu");
}

#[test]
fn pull_push_remote() {
    let ctx = TestContext::setup_clone();
    run(
        ctx.dir.path(),
        &["git", "config", "branch.main.pushRemote", "origin"],
    );

    snapshot!(ctx, "Fp");
}

#[test]
fn pull_upstream_prompt() {
    let ctx = TestContext::setup_clone();
    run(ctx.dir.path(), &["git", "branch", "--unset-upstream"]);
    snapshot!(ctx, "Fu");
}

#[test]
fn pull_push_remote_prompt() {
    let ctx = TestContext::setup_clone();
    snapshot!(ctx, "Fp");
}

#[test]
fn pull_setup_upstream() {
    let ctx = TestContext::setup_clone();
    run(ctx.dir.path(), &["git", "checkout", "-b", "new-branch"]);
    snapshot!(ctx, "Fumain<enter>F");
}

#[test]
fn pull_setup_upstream_same_as_head() {
    let ctx = TestContext::setup_clone();
    run(ctx.dir.path(), &["git", "checkout", "-b", "new-branch"]);
    snapshot!(ctx, "Funew-branch<enter>");
}

#[test]
fn pull_setup_push_remote() {
    let ctx = TestContext::setup_clone();
    snapshot!(ctx, "Fporigin<enter>F");
}

#[test]
fn pull_from_elsewhere_prompt() {
    snapshot!(TestContext::setup_clone(), "Fe");
}

#[test]
fn pull_from_elsewhere() {
    snapshot!(TestContext::setup_clone(), "Feorigin<enter>");
}
