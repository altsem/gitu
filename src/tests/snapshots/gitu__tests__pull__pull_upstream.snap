---
source: src/tests/pull.rs
expression: ctx.redact_buffer()
---
▌On branch main                                                                 |
▌Your branch is up to date with 'origin/main'.                                  |
                                                                                |
 Recent commits                                                                 |
 d07f2d3 main origin/main add remote-file                                       |
 b66a0bf add initial-file                                                       |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
────────────────────────────────────────────────────────────────────────────────|
$ git pull origin refs/heads/main                                               |
From                                                                            |
 * branch            main       -> FETCH_HEAD                                   |
   b66a0bf..d07f2d3  main       -> origin/main                                  |
Updating b66a0bf..d07f2d3                                                       |
Fast-forward                                                                    |
 remote-file | 1 +                                                              |
 1 file changed, 1 insertion(+)                                                 |
 create mode 100644 remote-file                                                 |
styles_hash: 5cab1b81bd346c16
